pallets/pool-system/src/impls.rs

```diff
@@ -367,6 +372,60 @@ impl<T: Config> InvestmentAccountant<T::AccountId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> ChangeGuard for Pallet<T> {
+	type Change = T::RuntimeChange;
+	type ChangeId = T::Hash;
+	type PoolId = T::PoolId;
+
+	fn note(pool_id: Self::PoolId, change: Self::Change) -> Result<Self::ChangeId, DispatchError> {
+		let registered_change = (Self::now(), change.clone());
+		let change_id: Self::ChangeId = T::Hashing::hash(&registered_change.encode());
+		Changes::<T>::insert(pool_id, change_id, registered_change);
+
+		Self::deposit_event(Event::ProposedChange {
+			pool_id,
+			change_id,
+			change,
+		});
+
+		Ok(change_id)
+	}
+
+	fn released(
+		pool_id: Self::PoolId,
+		change_id: Self::ChangeId,
+	) -> Result<Self::Change, DispatchError> {
+		let (submitted_time, change) =
+			Changes::<T>::get(pool_id, change_id).ok_or(Error::<T>::ChangeNotFound)?;
+
+		let pool_change: PoolChangeProposal = change.clone().into();
+
+		let mut allowed = true;
+		for requirement in pool_change.requirements() {
+			allowed &= match requirement {
+				Requirement::NextEpoch => {
+					match Pool::<T>::get(pool_id) {
+						Some(details) => submitted_time < details.epoch.last_closed,
+						None => true, // If the pool doesn't exists, there are no restrictions.
+					}
+				}
+				Requirement::DelayTime(secs) => {
+					Self::now().saturating_sub(submitted_time) >= secs as u64
+				}
+				// TODO
+				Requirement::BlockedByLockedRedemptions => false,
```

Comment:
I do not know how to know if this restriction is valid. Could it mean the we are blocked in the following interval?:
```
last_closed < submitted_time < last_executed
```

Reply:
See here how this restriction is implemented: https://github.com/centrifuge/centrifuge-chain/blob/main/runtime/centrifuge/src/lib.rs#L1323

Reply:
Actually we should also have that "do not allow releasing updates during epoch closing" check for all possible changes, both pool and loan.

Reply:
> Actually we should also have that "do not allow releasing updates during epoch closing" check for all possible changes, both pool and loan.

What do you mean by this? to have the `BlockedByLockedRedemptions` as a default implicit requirement for all changes?

Reply:
Ok, so we have a kind of default requirement for all possible changes

Reply:
Yes exactly. Should never allow releasing an update if the pool is in a submission period

Reply:
Done 👍🏻

Reply:
Looks good to me! Nice implementation :)